Deploy more smart pointers in WebProcess.cpp

rniwa opened the pull request:
#### 153a083bdd040be3159962114d3f337a86a4498a
<pre>
Deploy more smart pointers in WebProcess.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=260249">https://bugs.webkit.org/show_bug.cgi?id=260249</a>

Reviewed by Sihui Liu.

Deployed more smart pointers.

* Source/WebKit/WebProcess/WebCoreSupport/WebFileSystemStorageConnection.cpp:
(WebKit::WebFileSystemStorageConnection::create):
(WebKit::WebFileSystemStorageConnection::WebFileSystemStorageConnection):
* Source/WebKit/WebProcess/WebCoreSupport/WebFileSystemStorageConnection.h:
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::createWebPage):
(WebKit::WebProcess::handleInjectedBundleMessage):
(WebKit::WebProcess::setInjectedBundleParameter):
(WebKit::WebProcess::setInjectedBundleParameters):
(WebKit::WebProcess::ensureNetworkProcessConnection):
(WebKit::WebProcess::fileSystemStorageConnection):
(WebKit::WebProcess::ensureGPUProcessConnection):
(WebKit::WebProcess::remotePostMessage):
(WebKit::WebProcess::renderTreeAsText):
(WebKit::WebProcess::establishRemoteWorkerContextConnectionToNetworkProcess):
(WebKit::WebProcess::ensureSpeechRecognitionRealtimeMediaSourceManager):

Canonical link: <a href="https://commits.webkit.org/266952@main">https://commits.webkit.org/266952@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c125599f669d9ae7f1a08c983e6a66bd936fe80c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/15230 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15535 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15895 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/16986 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14307 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/18050 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15633 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/16918 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/15413 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15853 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/12950 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/17719 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/13133 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/20701 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14213 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/13906 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17164 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14475 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12268 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13750 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18094 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1842 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/14312 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebKit/WebProcess/WebProcess.cpp:
```diff
@@ -1311,7 +1311,7 @@ void WebProcess::networkProcessConnectionClosed(NetworkProcessConnection* connec
 {
     if (!m_fileSystemStorageConnection)
-        m_fileSystemStorageConnection = WebFileSystemStorageConnection::create(ensureNetworkProcessConnection().connection());
+        m_fileSystemStorageConnection = WebFileSystemStorageConnection::create(Ref { ensureNetworkProcessConnection().connection() });
```
Can you make `WebFileSystemStorageConnection` take Ref&& to avoid refcount churn?

rniwa replied:
Sure, fixed.

szewai reviewed (approved):
LGTM

rniwa committed:
Deploy more smart pointers in WebProcess.cpp
https://bugs.webkit.org/show_bug.cgi?id=260249

Reviewed by Sihui Liu.

Deployed more smart pointers.

* Source/WebKit/WebProcess/WebCoreSupport/WebFileSystemStorageConnection.cpp:
(WebKit::WebFileSystemStorageConnection::create):
(WebKit::WebFileSystemStorageConnection::WebFileSystemStorageConnection):
* Source/WebKit/WebProcess/WebCoreSupport/WebFileSystemStorageConnection.h:
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::createWebPage):
(WebKit::WebProcess::handleInjectedBundleMessage):
(WebKit::WebProcess::setInjectedBundleParameter):
(WebKit::WebProcess::setInjectedBundleParameters):
(WebKit::WebProcess::ensureNetworkProcessConnection):
(WebKit::WebProcess::fileSystemStorageConnection):
(WebKit::WebProcess::ensureGPUProcessConnection):
(WebKit::WebProcess::remotePostMessage):
(WebKit::WebProcess::renderTreeAsText):
(WebKit::WebProcess::establishRemoteWorkerContextConnectionToNetworkProcess):
(WebKit::WebProcess::ensureSpeechRecognitionRealtimeMediaSourceManager):

Canonical link: https://commits.webkit.org/266952@main
